Source/Aptabase/Private/AptabaseSettings.h

```diff
@@ -0,0 +1,63 @@
+﻿#pragma once
+
+#include <Engine/DeveloperSettings.h>
+
+#include "AptabaseSettings.generated.h"
+
+/**
+ * @brief Possible hosts for ingesting data. Some might require additional data (e.g.: SH - self hosted).
+ */
+UENUM()
+enum class EAptabaseHost : uint8
+{
+	EU,
+	US,
+	DEV,
+	SH
+};
+
+
+/**
+ * Holds configuration for integrating the Aptabase Analytics tracker
+ */
+UCLASS(Config = Aptabase)
+class APTABASE_API UAptabaseSettings : public UDeveloperSettings
+{
+	GENERATED_BODY()
+public:
+	/**
+	 * @brief Returns the base Url for all requests we will be sending out based on the Host
+	 */
+	FString GetApiUrl() const;
+	/**
+	 * @brief Key used to identify your app when making requests
+	 */
+	UPROPERTY(Config, EditAnywhere, Category = "Aptabase Analytics")
+	FString ApiKey = TEXT("");
+	/**
+	 * @brief Automatically determined based on the ApiKey. Will decide the
+	 */
+	UPROPERTY(Config, VisibleAnywhere, Category = "Aptabase Analytics")
+	EAptabaseHost Host;
+	/**
+	 * @brief Url to be used for self-hosted instances (Host = SH)
+	 */
+	UPROPERTY(Config, EditAnywhere, Category = "Aptabase Analytics", meta = (EditCondition = "Host == EAptabaseHost::SH", EditConditionHides))
+	FString CustomHost;
+	/**
+	 * @brief Whether this event should show up inside the "Release" or "Debug" dashboard inside the Web view
+	 */
+	UPROPERTY(Config, EditAnywhere, Category = "Aptabase Analytics")
+	bool bDebugMode = false;
```

Comment:
is this something that could be inferred instead of being a setting? maybe some sort or preprocessor like it's done in Swift

Reply:
It's actually the opposite, there are too many preprocessors and settings to choose from 😂.

So the usual application has 2 configurations: `Release` and `Debug`. So it makes sense to change those based on that.

However on the Unreal side we have 5 [Build configurations](https://docs.unrealengine.com/4.27/en-US/ProductionPipelines/DevelopmentSetup/BuildConfigurations/):
- Debug
- DebugGame
- Development
- Test
- Shipping

Most projects are distributed in `Shipping` mode, but there are products I worked on where people prefer to ship in `Test` because of the additional profiling capabilities.

For now, I think it's safe to consider `Shipping` to be `Release` and everything else to be `Debug` but we might change it in the future based on people's needs.

Reply:
Yeah, that sounds like a good approach. In future we could also let them choose when to consider it as Release/Debug, for having a good default like the one you mentioned is a good first step

Reply:
Awesome. I've adjusted the code and now only release games compiled in shipping are sending data as `Release`.

Based on the requests we get, we can adjust it to be more flexible solution, but this should be a great way to start things out.